src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -184,15 +184,19 @@ def handle_sso_command(cmd):
         if not sp_x_509_cert_path and sp_private_key_path:
             return -errno.EINVAL, '', 'Missing parameter `sp_x_509_cert`.'
         has_sp_cert = sp_x_509_cert_path != "" and sp_private_key_path != ""
-        try:
-            with open(sp_x_509_cert_path, 'r') as f:
-                sp_x_509_cert = f.read()
-        except FileNotFoundError:
+        if has_sp_cert:
+            try:
+                with open(sp_x_509_cert_path, 'r') as f:
+                    sp_x_509_cert = f.read()
+            except FileNotFoundError:
+                return -errno.EINVAL, '', '`{}` not found.'.format(sp_x_509_cert_path)
+            try:
+                with open(sp_private_key_path, 'r') as f:
+                    sp_private_key = f.read()
+            except FileNotFoundError:
+                return -errno.EINVAL, '', '`{}` not found.'.format(sp_private_key_path)
+        else:
```

Comment:
I didnt' realize that the `sp_private_key` and `sp_x_509_cert` were also used for either storing a filepath or file contents :S 

Give these are local files, I'd suggest to use `CephFilepath` in the `cmd` signature, which also performs file checks and deals with file permissions:
```diff
- 	             'name=sp_x_509_cert,type=CephString,req=false '
+ 	             'name=sp_x_509_cert,type=CephFilepath,req=false '
-                'name=sp_private_key,type=CephString,req=false',
+                'name=sp_private_key,type=CephFilepath,req=false',
```

Reply:
> I didnt' realize that the `sp_private_key` and `sp_x_509_cert` were also used for either storing a filepath or file contents :S

🤔 Isn't that an indication that the related unit/QA tests are not in a good state? Otherwise they would have uncovered the problem.

Reply:
@epuertat I'm now using `CephFilepath' instead of `CephString'.

Reply:
@ricardoasmarques great! Let see how that performs.

@votdev yeah, it seems so. Nevertheless, achieving 100% coverage on a method where arguments are interpreted differently according to their values (being a filepath, a URL or file contents) is always hard. That's why I suggested reducing the interface to a single form.